services/wallet/controllers_v3.go

```diff
@@ -717,3 +695,49 @@ func DisconnectCustodianLinkV3(s *Service) func(w http.ResponseWriter, r *http.R
 		return handlers.RenderContent(ctx, nil, w, http.StatusOK)
 	}
 }
+
+type CreateBraveWalletV4Request struct {
+	Geolocation string `json:"geolocation"`
+}
+
+// CreateBraveWalletV4 creates a brave wallet. This endpoint takes a geolocation as part of the request.
+func CreateBraveWalletV4(s *Service) func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+	return func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		verifier := httpsignature.ParameterizedKeystoreVerifier{
+			SignatureParams: httpsignature.SignatureParams{
+				Algorithm: httpsignature.ED25519,
+				Headers:   []string{"digest", "(request-target)"},
+			},
+			Keystore: &DecodeEd25519Keystore{},
+			Opts:     crypto.Hash(0),
+		}
+
+		// perform validation based on public key that the user submits
+		ctx, publicKey, err := verifier.VerifyRequest(r)
+		if err != nil {
+			return handlers.WrapError(err, "invalid http signature", http.StatusForbidden)
+		}
+
+		var c CreateBraveWalletV4Request
+		err = json.NewDecoder(r.Body).Decode(&c)
```

Comment:
Need a Validate method for this input, Geolocation should follow the ISO format.

Reply:
It will need to match whats in the list this will be redundant